### PR TITLE
[HttpFoundation] Allow setting session options via DSN

### LINKED ISSUE #41071: [Session] Not possible to set prefix/ttl in RedisSessionHandler settings via DSN
**Symfony version(s) affected**: 5.2.x

**Description**
The `RedisSessionHandler` allows to set two options: `prefix` and `ttl`.
Unfortunately, it looks like these settings are not properly based in the calling code:

https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/HttpFoundation/Session/Storage/Handler/SessionHandlerFactory.php#L53-L62

Instead, only the options `["lazy" => true]` are passed. The options from the DSN are not parsed for setting up the symfony-side of the config (they are only parsed to configure the redis client itself).

**How to reproduce**
Use the following DSN to configure your session client: `redis://localhost:6379?prefix=ohai_` and it will still use the default prefix `sf_s`.

The `SessionHandlerFactory` uses the redis parsing code from `symfony/cache`, so only the connection is set up with these settings and not the session handler itself.

**Possible Solution**
The only solution I see right now is to parse the query twice: once for setting up the session handler, once for creating the connection.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by makraz**
| Q             | A
| ------------- | ---
| Branch?       | 5.4 for bug fixes
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #41071
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Session/Storage/Handler/SessionHandlerFactory.php [hunk 1]
```diff
@@ -57,9 +57,28 @@ public static function createHandler($connection): AbstractSessionHandler
                     throw new \InvalidArgumentException(sprintf('Unsupported DSN "%s". Try running "composer require symfony/cache".', $connection));
                 }
                 $handlerClass = 0 === strpos($connection, 'memcached:') ? MemcachedSessionHandler::class : RedisSessionHandler::class;
+                $options = [];
+                if (RedisSessionHandler::class === $handlerClass) {
+                    if (false === $params = parse_url($connection)) {
+                        throw new InvalidArgumentException(sprintf('Invalid Redis DSN: "%s".', $connection));
```
Comment (SergeShin) [inline src/Symfony/Component/HttpFoundation/Session/Storage/Handler/SessionHandlerFactory.php]:
```suggestion
                        throw new \InvalidArgumentException(sprintf('Invalid Redis DSN: "%s".', $connection));
```

**Comment by nicolas-grekas** [comment]
Thanks for the PR. This looks like a new feature to me. As such, that's for 5.4.
Looking at other handlers, they would too benefit from options.
Also, `MemcachedSessionHandler` uses `expiretime` instead of `ttl` and defaults to 86400 instead of `ini_get('session.gc_maxlifetime')` :facepalm:
Can you please have a look and try to add some consistency there?
Don't forget to update the CHANGELOG of the component while doing so.
Thanks!

**Comment by makraz** [comment]
> Thanks for the PR. This looks like a new feature to me. As such, that's for 5.4.
> Looking at other handlers, they would too benefit from options.
> Also, `MemcachedSessionHandler` uses `expiretime` instead of `ttl` and defaults to 86400 instead of `ini_get('session.gc_maxlifetime')` 🤦
> Can you please have a look and try to add some consistency there?
> Don't forget to update the CHANGELOG of the component while doing so.
> Thanks!

@nicolas-grekas thank you for the feedback. I updated the PR to let other handlers benefit from options.

**Comment by nicolas-grekas** [comment]
@makraz I force-pushed some changes on your fork. Could you please fetch it and add some test cases?

**Comment by makraz** [comment]
> @makraz, I force-pushed some changes on your fork. Could you please fetch it and add some test cases?

Thank you, @nicolas-grekas, for the update. I will work on the test cases

**Comment by nicolas-grekas** [comment]
Friendly ping @makraz (rebase needed also)

**Comment by nicolas-grekas** [comment]
Thank you @makraz.

**Comment by nicolas-grekas** [comment]
Btw, PR adding tests + another adding doc would be really welcomed.

**Comment by makraz** [comment]
> Btw, PR adding tests + another adding doc would be really welcomed.

Thanks @nicolas-grekas for merging my PR, and sure I will handle it 🙌

**Pull request merged**